File: backend/mypersonalscheduler/mypersonalscheduler/src/main/java/com/kevinruedasv/mypersonalscheduler/service/UserService.java
package com.kevinruedasv.mypersonalscheduler.service;

import com.kevinruedasv.mypersonalscheduler.model.User;

public interface UserService {

    User register(
            String email,
            String username,
            String password
    );

    User login(
            String email,
            String password
    );

    User getUserById(
            String userId
    );
    
    User updateUsername(
            String userId,
            String username
    );

    void deleteUser(
            String userId
    );
}

File: frontend/src/app/features/profile/services/user.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { User } from '../../../core/models/user.model';
import { UpdateUserRequest } from '../models/update-user-request.model';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private readonly http = inject(HttpClient);

  private readonly API_URL = 'http://localhost:8080/api/users';

  getUser(userId: string): Observable<User> {
    return this.http.get<User>(`${this.API_URL}/${userId}`);
  }

  updateUsername(
    userId: string,
    data: UpdateUserRequest
  ): Observable<User> {
    return this.http.put<User>(
      `${this.API_URL}/${userId}`,
      data
    );
  }

  deleteUser(userId: string): Observable<void> {
    return this.http.delete<void>(
      `${this.API_URL}/${userId}`
    );
  }
}
